src/code/z_skin_awb.c

```diff
@@ -1,13 +1,203 @@
+    skin->avbTbl = NULL;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_80138424.s")
+/**
+ * Initialises the Vtx buffers used for limb at index `limbIndex`
+ */
+void Skin_InitAnimatedLimb(GameState* gameState, PSkinAwb* skin, s32 limbIndex) {
+    s32 i;
+    SkinLimb** skeleton = Lib_SegmentedToVirtual(skin->skeletonHeader->segment);
+    SkinAnimatedLimbData* animatedLimbData =
+        Lib_SegmentedToVirtual(((SkinLimb*)Lib_SegmentedToVirtual(skeleton[limbIndex]))->segment);
+    Struct_800A598C* temp_v0 = Lib_SegmentedToVirtual(animatedLimbData->unk_4);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_8013859C.s")
+    for (i = 0; i < ARRAY_COUNT(skin->avbTbl->buf); i++) {
+        Vtx* vertices = skin->avbTbl[limbIndex].buf[i];
+        Struct_800A598C* entry;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_80138700.s")
+        for (entry = temp_v0; entry < &temp_v0[animatedLimbData->unk_2]; entry++) {
+            SkinVtx* skinVertices = Lib_SegmentedToVirtual(entry->skinVertices);
+            SkinVtx* skinVtxEntry;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_801387D4.s")
+            for (skinVtxEntry = skinVertices; skinVtxEntry < &skinVertices[entry->skinVerticesCount];) {
+                Vtx* vtx = &vertices[skinVtxEntry->vtxIndex];
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_skin_awb/func_801388E4.s")
+                skinVtxEntry++;
+
+                vtx->n.flag = 0;
+                vtx->n.tc[0] = skinVtxEntry[-1].u;
+                vtx->n.tc[1] = skinVtxEntry[-1].v;
+                vtx->n.a = skinVtxEntry[-1].alpha;
+            }
+        }
+    }
+}
+
+/**
+ * Initializes a skin skeleton to looping animation, dynamically allocating the frame tables,
+ * and dynamically allocating and initializing the Vtx buffers for its animated limbs
+ */
+void Skin_Init(GameState* gameState, PSkinAwb* skin, SkeletonHeader* skeletonHeader, AnimationHeader* animationHeader) {
+    s32 limbCount;
+    s32 i;
+    SkinLimb** skeleton;
+    SkeletonHeader* virtSkelHeader = Lib_SegmentedToVirtual(skeletonHeader);
+
+    skin->avbCount = virtSkelHeader->limbCount;
+    skin->skeletonHeader = virtSkelHeader;
+    limbCount = skin->skeletonHeader->limbCount;
+
+    skeleton = Lib_SegmentedToVirtual(skin->skeletonHeader->segment);
+    skin->avbTbl = ZeldaArena_Malloc(limbCount * sizeof(SkinAvb));
+
+    for (i = 0; i < limbCount; i++) {
+        SkinAvb* avbEntry = &skin->avbTbl[i];
+
+        if ((((SkinLimb*)Lib_SegmentedToVirtual(skeleton[i]))->segmentType != SKIN_LIMB_TYPE_ANIMATED) ||
+            (((SkinLimb*)Lib_SegmentedToVirtual(skeleton[i]))->segment == NULL)) {
+            avbEntry->index = 0;
+
+            avbEntry->buf[0] = NULL;
+            avbEntry->buf[1] = NULL;
+        } else {
+            SkinAnimatedLimbData* animatedLimbData =
+                Lib_SegmentedToVirtual((((SkinLimb*)Lib_SegmentedToVirtual(skeleton[i]))->segment));
+
+            { s32 tmp; }
```

Comment:
```suggestion
        } else {
            s32 tmp;
            
            SkinAnimatedLimbData* animatedLimbData =
                Lib_SegmentedToVirtual((((SkinLimb*)Lib_SegmentedToVirtual(skeleton[i]))->segment));
```
Does this still match?

Reply:
No, the temp in the middle is required :c